taskvine/src/manager/vine_manager.c

```diff
@@ -4546,36 +4517,34 @@ int vine_submit(struct vine_manager *q, struct vine_task *t)
  * @param name The name of the library to be sent.
  * @return pointer to the library task if the operation succeeds, 0 otherwise.
  */
-
-static struct vine_task *send_library_to_worker(struct vine_manager *q, struct vine_worker_info *w, const char *name)
+struct vine_task *send_library_to_worker(struct vine_manager *q, struct vine_worker_info *w, const char *name)
 {
 	/* Find the original prototype library task by name, if it exists. */
 	struct vine_task *original = hash_table_lookup(q->libraries, name);
-	if (!original)
+	if (!original) {
 		return 0;
+	}
 
 	/*
 	If an instance of this library has recently failed,
 	don't send another right away.  This is a rather late (and inefficient)
 	point at which to notice this, but we don't know that we are starting
 	a new library until the moment of scheduling a function to that worker.
 	*/
-
 	timestamp_t lastfail = original->time_when_last_failure;
 	timestamp_t current = timestamp_get();
 	if (current < lastfail + q->transient_error_interval) {
 		return 0;
 	}
 
-	/* Duplicate the original task */
-	struct vine_task *t = vine_task_copy(original);
-
 	/* Check if this library task can fit in this worker. */
-	if (!check_worker_against_task(q, w, t)) {
-		vine_task_delete(t);
+	if (!check_worker_against_task(q, w, original)) {
```

Comment:
Hmm, the duplication of the task in the original version was inefficient, but it did provide some insurance against the possibility that `check_worker_against_task` might modify the original library object.    (For example, does `check_worker` do anything with regard to resource boxes?).   If you are 100% sure that this is the case, then we need some commentary (or maybe even use of `const`?) to ensure that tasks are not modified by `check_worker_against_task`.